Doxyfile.in

```diff
@@ -0,0 +1,14 @@
+PROJECT_NAME           = "@CMAKE_PROJECT_NAME@"
+PROJECT_NUMBER         = @VERSION_MAJOR@.@VERSION_MINOR@.@VERSION_PATCH@
+STRIP_FROM_PATH        = @PROJECT_SOURCE_DIR@ \
+                         @PROJECT_BINARY_DIR@
+INPUT                  = @doxy_main_page@ \
+                         @PROJECT_SOURCE_DIR@/src \
+                         @PROJECT_SOURCE_DIR@/include \
+                         @PROJECT_SOURCE_DIR@/doc
+FILE_PATTERNS          = *.hpp \
+                         *.inl \
+                         *.cpp \

```

Comment:
Because the documentation will be in the `.hpp`, we don't want to duplicate the documentation in the associated `.cpp` and I don't think that we want to declare functions, structures or classes in a `.cpp` except static functions and we don't want them to appear in the documentation because they can't be access from the outside of this file.